src/Stateless/StateConfiguration.Async.cs

```diff
@@ -21,7 +21,7 @@ public partial class StateConfiguration
             /// <returns></returns>
             public StateConfiguration InternalTransitionAsync(TTrigger trigger, Func<Transition, Task> entryAction)
             {
-                if (entryAction == null) throw new ArgumentNullException(nameof(entryAction));
+                Enforce.ArgumentNotNull(entryAction, nameof(entryAction));
```

Comment:
Given the built-in support for the `if (arg == null) throw ...` pattern in tools like ReSharper, it might be time to retire the `Enforce` pattern used in this lib. What do you think?